api/user.go

```diff
@@ -1,55 +1,70 @@
 package api
 
 import (
+	"context"
 	"encoding/json"
+	"fmt"
 	"memberserver/api/models"
 	"memberserver/config"
 	"memberserver/database"
 	"net/http"
-	"strings"
 	"time"
 
-	"github.com/dgrijalva/jwt-go"
+	"github.com/shaj13/go-guardian/v2/auth"
+	"github.com/shaj13/go-guardian/v2/auth/strategies/basic"
+	"github.com/shaj13/go-guardian/v2/auth/strategies/jwt"
+	"github.com/shaj13/go-guardian/v2/auth/strategies/union"
+	"github.com/shaj13/libcache"
+	_ "github.com/shaj13/libcache/fifo"
 	log "github.com/sirupsen/logrus"
 )
 
-// JWTExpireInterval - how long the JWT will last
+// JWTExpireInterval - how long the JWT will last in hours
 const JWTExpireInterval = 8
 
 // CookieName - name of the cookie :3
 const CookieName = "memberserver-token"
 
-// Create the JWT key used to create the signature
-var jwtKey = []byte("my_secret_key")
+var strategy union.Union
+var keeper jwt.SecretsKeeper
+
+func setupGoGuardian(config config.Config, db *database.Database) {
+
+	keeper = jwt.StaticSecret{
+		ID:        "secret-id",
```

Comment:
Would it be to create a `const` for the key id?

`const kid = "secret-id"`

Reply:
Sure, out of curiosity what is the advantage?

Reply:
It helps reduce the number of hard-coded values we have in the code base. So, if we plan to eventually change the `secret-id`  to something else, we will only have to do it in one place.

Reply:
While I typically agree with having constants defined, we don't use `"secret-id"` anywhere else.  So, I think it's fine to keep it how it is.

I'm a fan of minimizing how many variables there are.  Each variable you add adds a bit of cognitive load.

Reply:
Fair enough